src/res_printer.ml

```diff
@@ -705,19 +738,15 @@ and printModType ~customLayout modType cmtTbl =
           printLongidentLocation longident cmtTbl;
         ]
     | Pmty_signature [] ->
-      let shouldBreak =
-        modType.pmty_loc.loc_start.pos_lnum < modType.pmty_loc.loc_end.pos_lnum
-      in
-      Doc.breakableGroup ~forceBreak:shouldBreak
-        (Doc.concat
-           [
-             Doc.lbrace;
-             Doc.indent
-               (Doc.concat
-                  [Doc.softLine; printCommentsInside cmtTbl modType.pmty_loc]);
-             Doc.softLine;
-             Doc.rbrace;
-           ])
+      let doc = printCommentsInside cmtTbl modType.pmty_loc in
+      if Doc.isNil doc then
```

Comment:
This does the job but looks like tight coupling at a distance.
When is it that `printCommentsInside` currently produces `Nil`, and can we get hold of that information instead (e.g. return an optional).

Reply:
Yes, in an ideal world, we shouldn't inspect the contents of a doc to make formatting decisions. The information we go off is the AST (+ comments).

@ah-yu is the `isNil` absolutely necessary?

Reply:
`isNil` is only used in the development phase, then I could do a quick test without too many code changes.

Forgot to clean it up and will do it later.